### PR TITLE
Freeze pruning while creating snapshot

### PR DESCRIPTION
**Pull request by rakita**

### CODE REVIEW

File: ethcore/src/client/client.rs [hunk 1]
```diff
@@ -1152,16 +1156,27 @@ impl Client {
...
+                Some(earliest_era) if earliest_era + self.history <= latest_era => {
+                    let freeze_at = self.snapshotting_at.load(AtomicOrdering::SeqCst);
+                    if freeze_at > 0 && freeze_at == earliest_era {
+                        // Note: journal_db().mem_used() can be used for a more accurate memory
+                        // consumption measurement but it can be expensive so sticking with the
+                        // faster `journal_size()` instead.
+                        trace!(target: "pruning", "Pruning is paused at era {} (snapshot under way); earliest era={}, latest era={}, journal_size={} – Not pruning.",
+						       freeze_at, earliest_era, latest_era, state_db.journal_db().journal_size());
+                        break;
+                    }
```
Comment (rakita) [inline ethcore/src/client/client.rs:1168]: this is where we skip pruning if the snapshot is activated.

File: crates/ethcore/src/snapshot/service.rs [hunk 2]
```diff
@@ -529,45 +529,49 @@ impl Service {
...
 
-        let _ = fs::remove_dir_all(&temp_dir);
+            let _ = fs::remove_dir_all(&temp_dir);
 
-        let writer = LooseWriter::new(temp_dir.clone())?;
+            let writer = LooseWriter::new(temp_dir.clone())?;
 
-        let guard = Guard::new(temp_dir.clone());
-        let res = client.take_snapshot(writer, BlockId::Number(num), &self.progress);
-        self.taking_snapshot.store(false, Ordering::SeqCst);
```
Comment (rakita) [inline crates/ethcore/src/snapshot/service.rs:543]: taking_snapshot is moved to scopeguard

**Review by sunce86 — APPROVED** [review]
Looks good.

**Review by karim-agha — APPROVED** [review]
lgtm

**Pull request merged**